feat: Prepare audio, video, and live entries

benstov opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/kaltura/kmc-ng/222)
<!-- Reviewable:end -->

esakal commented:
Reviewed 31 of 31 files at r1.
Review status: all files reviewed at latest revision, 25 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvuW0LmVXWuYj5hfHDy:-KvuW0LnOyrNYQgTK5Ua:b-mun2wq) (no related file):*
+@amirch1 please review ui implementation and comment / lgtm

---

*[package-lock.json, line 0 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvuSOzU4u0d8q26vVX6:-KvuSOzU4u0d8q26vVX7:boipuag) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/package-lock.json#L0)):*
+@benstov this file shouldn't been update. Did you ran `npm install` of a new library?

I also don't see a merge commit so I assume you changed it somehow?

If so, please try to fix it by:
1. manual copy the one from `master`
2. delete `kmc-ng\node-modules` folder
3 run `npm install` in folder `kmc-ng`
4. test your app.

If it was on purpose please let me know the reason.

---

*[src/app-config/index.ts, line 1 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvuTFJm5GXquvtjt8Ea:-KvuTFJm5GXquvtjt8Eb:b52xq5n) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/app-config/index.ts#L1)):*
> ```XML
> export const environment = {
> ```

It seems that you re-formatted the document? was it on purpose?

we usually don't do it because it breaks the automatic merge and not relate to the feature being implemented. 

Since we currently don't enforce a specific rules (using `tslint` and `angular-cli` support) we have some differences in quotes, spaces etc.

Please try to avoid formatting documents that weren't created by you as part of the branch  

---

*[src/applications/kmc-upload-app/kmc-upload-app.module.ts, line 67 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvuVb9Kp0QR11MZb0ie:-KvuVb9Kp0QR11MZb0if:b-rv52di) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/kmc-upload-app.module.ts#L67)):*
> ```XML
>   ],
>   providers: [
>     PrepareEntryService,
> ```

setting a provider on a module makes it `singleton` which doesn't utilize the power of component DI and also enforces restrictions on the service state management.

Did you do it on purpose? what are the reasons for those services being part of the module?

---

*[src/applications/kmc-upload-app/create-live/create-live.component.scss, line 33 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvuYPC-LQrBIwF_uvy7:-KvuYPC-LQrBIwF_uvy8:bjxqpwf) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/create-live.component.scss#L33)):*
> ```SCSS
>   }
> 
>   /deep/ .kContainer {
> ```

what is the purpose of `deep` in this context? It doesn't seems to wrap global elements

---

*[src/applications/kmc-upload-app/create-live/create-live.component.scss, line 109 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvuYXeI04lO3ZIYMxEa:-KvuYXeI04lO3ZIYMxEb:b-o8t4yy) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/create-live.component.scss#L109)):*
> ```SCSS
> }
> 
> /deep/ {
> ```

defining a `/deep/` without a context is forbidden. use `:host /deep/` instead.

---

*[src/applications/kmc-upload-app/create-live/create-live.component.ts, line 10 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvuZB0Exa3ge_OZpLsR:-KvuZB0Exa3ge_OZpLsS:b2swx6o) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/create-live.component.ts#L10)):*
> ```XML
> import {PopupWidgetComponent, PopupWidgetStates} from '@kaltura-ng/kaltura-ui/popup-widget/popup-widget.component';
> 
> enum StreamTypes {
> ```

any declaration being part of the component public api must be exported as well. add `export` at the beginning 

---

*[src/applications/kmc-upload-app/create-live/create-live.component.ts, line 129 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-Kvu_TQzQ_A3-mX_26sx:-Kvu_TQzQ_A3-mX_26sy:b-7m74oy) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/create-live.component.ts#L129)):*
> ```XML
>       default: {
>         // add error message for trying to submit unsupported form type
>         const blockerMessage = new AreaBlockerMessage({
> ```

nice 

---

*[src/applications/kmc-upload-app/create-live/create-live.component.ts, line 134 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-Kvu_UN2arQrBF83U9iH:-Kvu_UN2arQrBF83U9iI:b-eu9h3k) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/create-live.component.ts#L134)):*
> ```XML
>           buttons: [{
>             label: this._appLocalization.get('app.common.confirm'),
>             action: () => {}
> ```

was this tested? because you are not closing the blocker so I think the user will not be able to continue and choose something else.... 

---

*[src/applications/kmc-upload-app/create-live/create-live.service.ts, line 40 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvuXQsYAOygc03lEGPp:-KvuXQsYAOygc03lEGPq:by2e5ml) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/create-live.service.ts#L40)):*
> ```XML
> }
> 
> export {KalturaLive, ManualLive, UniversalLive}
> ```

you should export each element separately.

[see example](https://github.com/kaltura/kmc-ng/blob/f4f9c32d72a18dd8bbff3e795af08997a920ae4c/src/applications/content-entries-app/entry/entry-thumbnails/entry-thumbnails-handler.ts#L28)

---

*[src/applications/kmc-upload-app/create-live/kaltura-live-stream/kaltura-live-stream.component.ts, line 28 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvubS4JszivPf18fBmH:-KvubS4JszivPf18fBmI:b-fg4m1f) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/kaltura-live-stream/kaltura-live-stream.component.ts#L28)):*
> ```XML
> 
>   @Input()
>   blockerState: { isBusy: boolean, message: AreaBlockerMessage };
> ```

what benefit do you have blocking the parent instead off having a separated instance in the component?

---

*[src/applications/kmc-upload-app/create-live/kaltura-live-stream/kaltura-live-stream.component.ts, line 42 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvubjuXoqxFajo5yTgm:-KvubjuXoqxFajo5yTgn:bs5pbpv) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/kaltura-live-stream/kaltura-live-stream.component.ts#L42)):*
> ```XML
>     this._createForm();
>     this._fillEnableRecordingOptions();
>     setTimeout(() => {
> ```

what happens without the setTimeout?

---

*[src/applications/kmc-upload-app/create-live/kaltura-live-stream/kaltura-live-stream.component.ts, line 77 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvulBUHY8wlhp4DX_96:-KvulBUHY8wlhp4DX_97:b-ji15iq) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/kaltura-live-stream/kaltura-live-stream.component.ts#L77)):*
> ```XML
> 
>       }, error => {
>         this._updateAreaBlockerState(false, error.message);
> ```

I'm not sure how this is working because the error.message is string and you ask for `AreaBlockerMessage`. we created a [separated task](https://kaltura.atlassian.net/browse/KMCNG-774) to correctly support message as string. in the meanwhile just fix the declaration of the function `_updateAreaBlockerState`

---

*[src/applications/kmc-upload-app/create-live/kaltura-live-stream/kaltura-live-stream.component.ts, line 90 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-Kvuo4p0lMzJJXrcsQ9i:-Kvuo4p0lMzJJXrcsQ9j:b16bzjl) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/kaltura-live-stream/kaltura-live-stream.component.ts#L90)):*
> ```XML
>       return profileIdFromCache;
>     } else {
>       this._browserService.setInLocalStorage('kalturaStreamType.selectedTranscodingProfile', transcodingProfilesList[0].id);
> ```

[minor] protect against empty transcoding list

---

*[src/applications/kmc-upload-app/create-live/kaltura-live-stream/kaltura-live-stream.component.ts, line 140 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-Kvukv95z83jWBu5EAQd:-Kvukv95z83jWBu5EAQe:bworjme) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/kaltura-live-stream/kaltura-live-stream.component.ts#L140)):*
> ```XML
> 
>   private _updateAreaBlockerState(isBusy: boolean, message: AreaBlockerMessage): void {
>     this.blockerStateChange.emit({isBusy, message})
> ```

hmm... is this working? because I don't see in the parent component any place where you intercept and handle this change

---

*[src/applications/kmc-upload-app/create-live/kaltura-live-stream/kaltura-live-stream.service.ts, line 13 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-Kvv550gJQVjeUtfpuHb:-Kvv550gJQVjeUtfpuHc:b-64u6sk) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/kaltura-live-stream/kaltura-live-stream.service.ts#L13)):*
> ```XML
> export class KalturaLiveStreamService {
> 
>   private _cachedConversionProfiles: KalturaConversionProfile[] = null;
> ```

following our discussion:
- we should try to avoid having a service cache when the service is singleton to the application (because we don't handle updates of that data from within the application)
- for this specific implementation you can remove the caching
- check if can use/extend `TranscodingProfileManagement` if so use it 

+@amirch1 fyi

---

*[src/applications/kmc-upload-app/create-live/kaltura-live-stream/kaltura-live-stream.service.ts, line 21 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-Kvv6zyMwVZGgP1SOvl7:-Kvv6zyMwVZGgP1SOvl8:b-twr3nr) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/kaltura-live-stream/kaltura-live-stream.service.ts#L21)):*
> ```XML
>   public getKalturaConversionProfiles(): Observable<KalturaConversionProfile[]> {
>     return Observable.create(observer => {
>       if (!this._cachedConversionProfiles) {
> ```

for you future knowledge, this is not a solid way to implement caching when using observables. We can discuss about it later, in the meanwhile take a look how it was done by Stas [here](https://github.com/kaltura/kmc-ng/blob/e4d7ae577e893de905137acebfa5b96c09d4c338/src/shared/content-shared/categories-store.service.ts#L108)

+@amirch1 fyi

---

*[src/applications/kmc-upload-app/create-live/universal-live/universal-live.service.ts, line 37 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvvAH9r_sV6YRh2NyLo:-KvvAH9r_sV6YRh2NyLp:b3kre) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/universal-live/universal-live.service.ts#L37)):*
> ```XML
> 
>   private _getDefaultIp(): Observable<string> {
>     const akamaiEdgeServerIpURL = environment.modules.createLive.akamaiEdgeServerIpURL;
> ```

?

---

*[src/applications/kmc-upload-app/create-live/universal-live/universal-live.service.ts, line 38 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvvFMW_Nw0WZMufQkrN:-KvvFMW_Nw0WZMufQkrO:b-waje3p) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/universal-live/universal-live.service.ts#L38)):*
> ```XML
>   private _getDefaultIp(): Observable<string> {
>     const akamaiEdgeServerIpURL = environment.modules.createLive.akamaiEdgeServerIpURL;
>     return this._http.get(window.location.protocol + '//' + akamaiEdgeServerIpURL)
> ```

since it is not intuitive to ask for a url from configuration without protocol please add here a logic guarding from getting protocol from configuration which will lead to `https://http://kalvodhds-i.akamaihd.net/serverIp`

you can handle it silently, just remove the protocol from the value retrieved from the configuration if any and add the one from window.

---

*[src/applications/kmc-upload-app/prepare-entry/prepare-entry.component.ts, line 25 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvvAivs_kPrGztUMSW8:-KvvAivs_kPrGztUMSW9:bdwqcbu) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/prepare-entry/prepare-entry.component.ts#L25)):*
> ```XML
>   public prepareEntry(kalturaMediaType: KalturaMediaType) {
>     this._selectedMediaType = kalturaMediaType;
>     // TODO [kmcng] If user permissions allows setting transcoding profile - show transcoding profile selector
> ```

remind me where is the temp flag to test how it works when user doesn't have the right permission?

---

*[src/applications/kmc-upload-app/prepare-entry/prepare-entry.component.ts, line 36 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvvBQNUdjzq2F_6DnYM:-KvvBQNUdjzq2F_6DnYN:b9vq7nu) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/prepare-entry/prepare-entry.component.ts#L36)):*
> ```XML
> 
>     this._prepareEntryService.createDraftEntry(this._selectedMediaType,
>       selectedProfile.profileId || -1)
> ```

did you mean `selectedProfile? selectedProfile.profileId : -1` ? but since it is optional in `createDraftEntry` then it seems you should assign `null` instead of having two places that are aware to the `-1` default value.

---

*[src/applications/kmc-upload-app/prepare-entry/prepare-entry.service.ts, line 32 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvvC2FGbBnyBEe4HCSA:-KvvC2FGbBnyBEe4HCSB:b-tih4qt) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/prepare-entry/prepare-entry.service.ts#L32)):*
> ```XML
>     return this._kalturaServerClient
>       .request(new MediaAddAction({entry}))
>       .map(media => ({id: media.id}));
> ```

you are not "friendly" handling server errors

consider showing a friendlier message "failed to create entry" with possible "retry button"...

+@amirch1 what do you think?

![✂-1](https://files.reviewableusercontent.io/images/reviewable/4751797/WqkTdWij1i5JL20c2VoDS4k9.png)

---

*[src/applications/kmc-upload-app/prepare-entry/transcoding-profile-select/transcoding-profile-select.component.scss, line 49 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvvDDFu5vEKmUY3mB-8:-KvvDDFu5vEKmUY3mB-9:bhr980u) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/prepare-entry/transcoding-profile-select/transcoding-profile-select.component.scss#L49)):*
> ```SCSS
> }
> 
> /deep/ {
> ```

see other comment about `/deep/` not scoped to `:host` - also fix other places if you have such.

---

*[src/applications/kmc-upload-app/prepare-entry/transcoding-profile-select/transcoding-profile-select.component.ts, line 61 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvvDVw-iGwmxRcyUwQU:-KvvDVw-iGwmxRcyUwQV:b3iq35q) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/prepare-entry/transcoding-profile-select/transcoding-profile-select.component.ts#L61)):*
> ```XML
> 
>           // get profile from local cache if exists
>           const cachedSelectedProfile = this._browserService.getFromLocalStorage('transcodingProfiles.selectedProfile');
> ```

consider handling scenario where the value not exists in cache or the value is not found in the list (either removed from account or the user logged with different partner in the same machine)

---

*[src/applications/kmc-upload-app/prepare-entry/transcoding-profile-select/transcoding-profile-select.component.ts, line 89 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvvDjNlWfv-27sRkVfe:-KvvDjNlWfv-27sRkVff:bu21lno) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/prepare-entry/transcoding-profile-select/transcoding-profile-select.component.ts#L89)):*
> ```XML
>   private _saveAndClose() {
>     // set profile in local cache
>     const selectedProfile = this.transcodingProfileSelectForm.get('profile').value;
> ```

since we are using the primeng dropdown which might break itself in the future, it is adviced to validate if selectedProfile is not null. obviously we don't need to be overprotective and make sure the value if from the provided list (this will be an overkill) but let's protect from nullable.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/222)*
<!-- Sent from Reviewable.io -->

amirch1 commented:
Reviewed 31 of 31 files at r1.
Review status: all files reviewed at latest revision, 26 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvzY4DK6rNgZ3Kz4Um-:-KvzY4DK6rNgZ3Kz4Um0:b-nxbwz8) (no related file):*
>  prepare-audio-video-live
1. Getting some compilation warnings upon build.
2. Once a draft entry is created and returning back to the entries list, you need to refresh the list in order to see the new entry. Lets try to refresh it automatically if possible in this situation.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/222)*
<!-- Sent from Reviewable.io -->

amirch1 commented:
Review status: all files reviewed at latest revision, 27 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvzYtbgA6-xc6hWByTh:-KvzYtbgA6-xc6hWByTi:b-2g99eo) (no related file):*
Switch button has some layout and styling issues. Also, some labels are multi-lines - lets try and have the labels in a single line

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/222)*
<!-- Sent from Reviewable.io -->

benstov commented:
Review status: 14 of 36 files reviewed at latest revision, 27 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvuW0LmVXWuYj5hfHDy:-Kw5J1BEgtbQZ-3jKq3m:b-7rl0jd) (no related file):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

+@amirch1 please review ui implementation and comment / lgtm
</blockquote></details>

done.

---

*[a discussion](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvzY4DK6rNgZ3Kz4Um-:-Kw5J-bCGOYxUVFA2Y23:b-7rl0jd) (no related file):*
<details><summary><i>Previously, amirch1 (Amir Chervinsky) wrote…</i></summary><blockquote>

>  prepare-audio-video-live
1. Getting some compilation warnings upon build.
2. Once a draft entry is created and returning back to the entries list, you need to refresh the list in order to see the new entry. Lets try to refresh it automatically if possible in this situation.
</blockquote></details>

done.

---

*[a discussion](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvzYtbgA6-xc6hWByTh:-Kw4PtukvWumtsseJi6L:b-rnraql) (no related file):*
<details><summary><i>Previously, amirch1 (Amir Chervinsky) wrote…</i></summary><blockquote>

Switch button has some layout and styling issues. Also, some labels are multi-lines - lets try and have the labels in a single line
</blockquote></details>

done.

---

*[src/applications/kmc-upload-app/create-live/create-live.component.ts, line 134 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-Kvu_UN2arQrBF83U9iH:-Kw50WesMZ_3e6eytlNl:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/create-live.component.ts#L134)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

was this tested? because you are not closing the blocker so I think the user will not be able to continue and choose something else.... 
</blockquote></details>

Done.

---

*[src/applications/kmc-upload-app/create-live/kaltura-live-stream/kaltura-live-stream.component.ts, line 42 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvubjuXoqxFajo5yTgm:-Kw5J4Ut0yOM7bxilvoP:bb3do54) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/kaltura-live-stream/kaltura-live-stream.component.ts#L42)):*
<details><summary><i>Previously, benstov wrote…</i></summary><blockquote>

i put it there since i get an error `ExpressionChangedAfterItHasBeenCheckedError`, and if i want to have the         this._updateAreaBlockerState(false, null); to work so i have to put it inside setTimeout
</blockquote></details>

Removed the setTimeout since the areaBlocker state no longer exists  

---

*[src/applications/kmc-upload-app/create-live/kaltura-live-stream/kaltura-live-stream.component.ts, line 77 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvulBUHY8wlhp4DX_96:-Kw5JVLxeE1nbNwXDjhp:b-qssebl) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/kaltura-live-stream/kaltura-live-stream.component.ts#L77)):*
<details><summary><i>Previously, benstov wrote…</i></summary><blockquote>

> AreaBlockerMessage
export class AreaBlockerComponent implements OnInit  {

  public _message : AreaBlockerMessage;

  @Input() showLoader : boolean;

  @Input()
  set message(value : AreaBlockerMessage | string)
  {
    if (typeof value === 'string')
    {
      this._message = { title : 'Uh oh!', message : value, buttons : [{ label :'Dismiss', action : () => { this._message = null;}}]};
    }else if (value instanceof AreaBlockerMessage)
    {
      this._message = value;
    }else
    {
      this._message = null;
    }
  }

</blockquote></details>

fixed.

---

*[src/applications/kmc-upload-app/create-live/universal-live/universal-live.service.ts, line 37 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvvAH9r_sV6YRh2NyLo:-Kw4F2TSNjTKCfSaHWmU:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/universal-live/universal-live.service.ts#L37)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

?
</blockquote></details>

Done.

---

*[src/applications/kmc-upload-app/create-live/universal-live/universal-live.service.ts, line 38 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvvFMW_Nw0WZMufQkrN:-Kw4F34pIn3t4nbTUgJk:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/universal-live/universal-live.service.ts#L38)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

since it is not intuitive to ask for a url from configuration without protocol please add here a logic guarding from getting protocol from configuration which will lead to `https://http://kalvodhds-i.akamaihd.net/serverIp`

you can handle it silently, just remove the protocol from the value retrieved from the configuration if any and add the one from window.
</blockquote></details>

Done.

---

*[src/applications/kmc-upload-app/prepare-entry/prepare-entry.component.ts, line 25 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvvAivs_kPrGztUMSW8:-Kw4FaVb5HeNwSoJJBms:b-l910oh) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/prepare-entry/prepare-entry.component.ts#L25)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

remind me where is the temp flag to test how it works when user doesn't have the right permission?
</blockquote></details>

added the flag now, i have tested it before but didn't leave the temp flag.

---

*[src/applications/kmc-upload-app/prepare-entry/prepare-entry.component.ts, line 36 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvvBQNUdjzq2F_6DnYM:-Kw4GlQuNUFes8O0FonZ:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/prepare-entry/prepare-entry.component.ts#L36)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

did you mean `selectedProfile? selectedProfile.profileId : -1` ? but since it is optional in `createDraftEntry` then it seems you should assign `null` instead of having two places that are aware to the `-1` default value.
</blockquote></details>

Done.

---

*[src/applications/kmc-upload-app/prepare-entry/prepare-entry.service.ts, line 32 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvvC2FGbBnyBEe4HCSA:-Kw4HPPnFztPMP7jDmQT:b-xaw86o) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/prepare-entry/prepare-entry.service.ts#L32)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

you are not "friendly" handling server errors

consider showing a friendlier message "failed to create entry" with possible "retry button"...

+@amirch1 what do you think?

![✂-1](https://files.reviewableusercontent.io/images/reviewable/4751797/WqkTdWij1i5JL20c2VoDS4k9.png)

</blockquote></details>

Unable to set the buttons inside 'browserService.setAppStatus' since
 AppStatus.errorMessage type is a string and not AreaBlockerMessage.
Need to refactor the app.component.ts in order to support altering the '_blockerMessage'

---

*[src/applications/kmc-upload-app/prepare-entry/transcoding-profile-select/transcoding-profile-select.component.scss, line 49 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvvDDFu5vEKmUY3mB-8:-Kw4FuZxZKn6Om8VBAGD:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/prepare-entry/transcoding-profile-select/transcoding-profile-select.component.scss#L49)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

see other comment about `/deep/` not scoped to `:host` - also fix other places if you have such.
</blockquote></details>

Done.

---

*[src/applications/kmc-upload-app/prepare-entry/transcoding-profile-select/transcoding-profile-select.component.ts, line 61 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvvDVw-iGwmxRcyUwQU:-Kw4P4kSSH4G8pBwshc3:b-5zqzh2) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/prepare-entry/transcoding-profile-select/transcoding-profile-select.component.ts#L61)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

consider handling scenario where the value not exists in cache or the value is not found in the list (either removed from account or the user logged with different partner in the same machine)
</blockquote></details>

added _getSelectedTranscodingProfile function

---

*[src/applications/kmc-upload-app/prepare-entry/transcoding-profile-select/transcoding-profile-select.component.ts, line 89 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvvDjNlWfv-27sRkVfe:-Kw4PglcIKm7CF5uZwp8:b7t5iy2) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/prepare-entry/transcoding-profile-select/transcoding-profile-select.component.ts#L89)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

since we are using the primeng dropdown which might break itself in the future, it is adviced to validate if selectedProfile is not null. obviously we don't need to be overprotective and make sure the value if from the provided list (this will be an overkill) but let's protect from nullable.
</blockquote></details>

but it can be nullable, and if so , it will be assigned with -1 in the service

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/222)*
<!-- Sent from Reviewable.io -->

esakal commented:
Reviewed 22 of 22 files at r2.
Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

*[src/applications/content-entries-app/entry/entry-store.service.ts, line 87 at r2](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KweKIZfoyMtADJ5cbXP:-KweKIZfoyMtADJ5cbXQ:bg84yf2) ([raw file](https://github.com/kaltura/kmc-ng/blob/da4ac961a639b9a7ce1f20588573148bab9d5ba8/src/applications/content-entries-app/entry/entry-store.service.ts#L87)):*
> ```XML
>         this._entryRoute.queryParams.cancelOnDestroy(this)
>           .subscribe(queryParams => {
>             this._saveEntryInvoked = !!queryParams['reloadEntries']; // convert to boolean
> ```

- You should assign `true` only, setting it to false might affect scenarios where we set this flag and this code changes it to false.
- since you extended the purpose of this flag, consider using a better name like `_reloadEntiesListOnNavigateOut`
- consider using rxjs `.first()` operator since you just want to handle this flag once.

---

*[src/applications/kmc-upload-app/create-live/create-live.component.scss, line 33 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvuYPC-LQrBIwF_uvy7:-KweG6EiivoAt2aGj_dj:b-xm85r1) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/create-live.component.scss#L33)):*
<details><summary><i>Previously, benstov wrote…</i></summary><blockquote>

i wanted that this scss file will affect only children components, i didn't know that it must be concatenated with the ":host" selector
</blockquote></details>

but `kContainer` is a class of in this component html, did you try removing it?

---

*[src/applications/kmc-upload-app/create-live/create-live.component.ts, line 173 at r2](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KweGbQh6FFXYLbKQbKW:-KweGbQijcgS1t_cefsy:b-7s4s8d) ([raw file](https://github.com/kaltura/kmc-ng/blob/da4ac961a639b9a7ce1f20588573148bab9d5ba8/src/applications/kmc-upload-app/create-live/create-live.component.ts#L173)):*
> ```XML
>         message: this._appLocalization.get('applications.upload.prepareLive.confirmEntryNavigation.message'),
>         accept: () => {
>           this._router.navigate(['/content/entries/entry', id], {queryParams: {reloadEntries: true}})
> ```

cool!

---

*[src/applications/kmc-upload-app/create-live/create-live.service.ts, line 12 at r2](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KweGihCxwTepCdL-E9r:-KweGihCxwTepCdL-E9s:bpl93uy) ([raw file](https://github.com/kaltura/kmc-ng/blob/da4ac961a639b9a7ce1f20588573148bab9d5ba8/src/applications/kmc-upload-app/create-live/create-live.service.ts#L12)):*
> ```XML
> import {KalturaLiveStreamConfiguration} from 'kaltura-typescript-client/types/KalturaLiveStreamConfiguration';
> import {KalturaPlaybackProtocol} from 'kaltura-typescript-client/types/KalturaPlaybackProtocol';
> import {KalturaLive} from './kaltura-live-stream/kaltura-live-stream.interface';
> ```

well sorry for the confusion, it wasn't my intention. It is reasonable having the interfaces together with the services, it is also easier to maintain and follow.

When I wrote that you should export each one of them I meants just adding the keyword `export` before each interface instead of 

```
export {KalturaLive, ManualLive, UniversalLive}
```

---

*[src/applications/kmc-upload-app/prepare-entry/prepare-entry.component.ts, line 25 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvvAivs_kPrGztUMSW8:-KweI32qRG7vJ4QdKMPM:b-p041o1) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/prepare-entry/prepare-entry.component.ts#L25)):*
<details><summary><i>Previously, benstov wrote…</i></summary><blockquote>

added the flag now, i have tested it before but didn't leave the temp flag.
</blockquote></details>

actually I don't understand how this works if the flag return `false` because I don't see where you invoke the process automatically.

---

*[src/applications/kmc-upload-app/prepare-entry/prepare-entry.service.ts, line 32 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvvC2FGbBnyBEe4HCSA:-KweIYDUBRkFhDuY4sDv:b-njdx6o) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/prepare-entry/prepare-entry.service.ts#L32)):*
<details><summary><i>Previously, benstov wrote…</i></summary><blockquote>

Unable to set the buttons inside 'browserService.setAppStatus' since
 AppStatus.errorMessage type is a string and not AreaBlockerMessage.
Need to refactor the app.component.ts in order to support altering the '_blockerMessage'
</blockquote></details>

still you can manipulate the content of the message to a more destrictive one. ignore the 'retry' part.

---

*[src/applications/kmc-upload-app/prepare-entry/transcoding-profile-select/transcoding-profile-select.component.ts, line 89 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvvDjNlWfv-27sRkVfe:-KweJVQkbee8-NGRwqJW:b1bjj40) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/prepare-entry/transcoding-profile-select/transcoding-profile-select.component.ts#L89)):*
<details><summary><i>Previously, benstov wrote…</i></summary><blockquote>

but it can be nullable, and if so , it will be assigned with -1 in the service
</blockquote></details>

since it can be nullable you will fail here. 

```
this.transcodingProfileSelectForm.get('profile').value;
                                                                                ^---------- undefined if no profile selected - will throw unhandled error
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/222)*
<!-- Sent from Reviewable.io -->

benstov commented:
Review status: 31 of 36 files reviewed at latest revision, 9 unresolved discussions.

---

*[src/applications/content-entries-app/entry/entry-store.service.ts, line 87 at r2](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KweKIZfoyMtADJ5cbXP:-KxBtM5vL-jhx8RfV7dm:b-qssecl) ([raw file](https://github.com/kaltura/kmc-ng/blob/da4ac961a639b9a7ce1f20588573148bab9d5ba8/src/applications/content-entries-app/entry/entry-store.service.ts#L87)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

- You should assign `true` only, setting it to false might affect scenarios where we set this flag and this code changes it to false.
- since you extended the purpose of this flag, consider using a better name like `_reloadEntiesListOnNavigateOut`
- consider using rxjs `.first()` operator since you just want to handle this flag once.
</blockquote></details>

fixed

---

*[src/applications/kmc-upload-app/create-live/create-live.component.scss, line 33 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvuYPC-LQrBIwF_uvy7:-Kx8WzETx4-cU3xixBhY:b-f2j017) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/create-live.component.scss#L33)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

but `kContainer` is a class of in this component html, did you try removing it?
</blockquote></details>

We have discussed about it, and here it's ok to leave it as is (i understood the difference between that and leaving the /deep/ without parent)

---

*[src/applications/kmc-upload-app/create-live/create-live.component.ts, line 173 at r2](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KweGbQh6FFXYLbKQbKW:-Kx8-ofCmSAAvFAgzngs:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/da4ac961a639b9a7ce1f20588573148bab9d5ba8/src/applications/kmc-upload-app/create-live/create-live.component.ts#L173)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

cool!
</blockquote></details>

Done.

---

*[src/applications/kmc-upload-app/create-live/create-live.service.ts, line 12 at r2](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KweGihCxwTepCdL-E9r:-Kx8YOJ4O1IGJPxxZPRq:b3arszs) ([raw file](https://github.com/kaltura/kmc-ng/blob/da4ac961a639b9a7ce1f20588573148bab9d5ba8/src/applications/kmc-upload-app/create-live/create-live.service.ts#L12)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

well sorry for the confusion, it wasn't my intention. It is reasonable having the interfaces together with the services, it is also easier to maintain and follow.

When I wrote that you should export each one of them I meants just adding the keyword `export` before each interface instead of 

```
export {KalturaLive, ManualLive, UniversalLive}
```
</blockquote></details>

I think i will leave it like i did since i am getting warnings

---

*[src/applications/kmc-upload-app/prepare-entry/prepare-entry.component.ts, line 25 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvvAivs_kPrGztUMSW8:-KxCLP6723QgHQObN3Dx:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/prepare-entry/prepare-entry.component.ts#L25)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

actually I don't understand how this works if the flag return `false` because I don't see where you invoke the process automatically.
</blockquote></details>

Done.

---

*[src/applications/kmc-upload-app/prepare-entry/prepare-entry.service.ts, line 32 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvvC2FGbBnyBEe4HCSA:-KxCNIDNESe_PMmLKq3p:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/prepare-entry/prepare-entry.service.ts#L32)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

still you can manipulate the content of the message to a more destrictive one. ignore the 'retry' part.
</blockquote></details>

Done.

---

*[src/applications/kmc-upload-app/prepare-entry/transcoding-profile-select/transcoding-profile-select.component.ts, line 89 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvvDjNlWfv-27sRkVfe:-KxCJ7vjzILKmQGpOX60:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/prepare-entry/transcoding-profile-select/transcoding-profile-select.component.ts#L89)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

since it can be nullable you will fail here. 

```
this.transcodingProfileSelectForm.get('profile').value;
                                                                                ^---------- undefined if no profile selected - will throw unhandled error
```
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/222)*
<!-- Sent from Reviewable.io -->

esakal commented:
Reviewed 5 of 5 files at r3.
Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

*[src/applications/kmc-upload-app/create-live/create-live.component.ts, line 173 at r2](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KweGbQh6FFXYLbKQbKW:-KxI7eEuvicTzg2SE-AZ:b-vhks1y) ([raw file](https://github.com/kaltura/kmc-ng/blob/da4ac961a639b9a7ce1f20588573148bab9d5ba8/src/applications/kmc-upload-app/create-live/create-live.component.ts#L173)):*
<details><summary><i>Previously, benstov wrote…</i></summary><blockquote>

Done.
</blockquote></details>

if you can remove the prefix '_' because this is considered as public api

---

*[src/applications/kmc-upload-app/prepare-entry/prepare-entry.component.ts, line 31 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KxI7ubv0_6iAKvkFpcd:-KxI7ubv0_6iAKvkFpce:b-coj9o0) ([raw file](https://github.com/kaltura/kmc-ng/blob/8e658724b3d8dce61fb2e99dd7d7cef996e521e2/src/applications/kmc-upload-app/prepare-entry/prepare-entry.component.ts#L31)):*
> ```XML
>     // 'transcodingProfileSettingPermission' should contain whether the user has the permission to set the transcoding profile
>     const transcodingProfileSettingPermission = false;
>     if (transcodingProfileSettingPermission) {  this.transcodingProfileSelectMenu.open();
> ```

isn't this a duplication with the line below?

---

*[src/applications/kmc-upload-app/prepare-entry/prepare-entry.component.ts, line 48 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KxI8-edtEsroREE4MYC:-KxI8-edtEsroREE4MYD:b-frvyko) ([raw file](https://github.com/kaltura/kmc-ng/blob/8e658724b3d8dce61fb2e99dd7d7cef996e521e2/src/applications/kmc-upload-app/prepare-entry/prepare-entry.component.ts#L48)):*
> ```XML
>     this._prepareEntryService.createDraftEntry(this._selectedMediaType, selectedProfile.profileId)
>       .subscribe((draftEntry: DraftEntry) => {
>           this._router.navigate(['/content/entries/entry', draftEntry.id], {queryParams: {_reloadEntiesListOnNavigateOut: true}})
> ```

the same comment as above about the prefix of _

---

*[src/applications/kmc-upload-app/prepare-entry/transcoding-profile-select/transcoding-profile-select.component.ts, line 89 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvvDjNlWfv-27sRkVfe:-KxI8P2y15GJi2hoUpjZ:b-82l5ff) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/prepare-entry/transcoding-profile-select/transcoding-profile-select.component.ts#L89)):*
<details><summary><i>Previously, benstov wrote…</i></summary><blockquote>

Done.
</blockquote></details>

you will fail right after the `.value` part....  

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/222)*
<!-- Sent from Reviewable.io -->

amirch1 commented:
Reviewed 18 of 22 files at r2, 5 of 5 files at r3.
Review status: 32 of 36 files reviewed at latest revision, 8 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KxICCHI-Um5OUUPVBMK:-KxICCHI-Um5OUUPVBML:b-qlmqgo) (no related file):*
After creating a live entering, if the user chooses not to view the entry details, the screen is stuck on the floater and closing it results in "data lost" confirmation. Better to return to the entries list and refresh it so the user will be able to see the newly created entry

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/222)*
<!-- Sent from Reviewable.io -->

esakal commented:
Reviewed 4 of 4 files at r4.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[src/applications/kmc-upload-app/create-live/create-live.component.scss, line 33 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvuYPC-LQrBIwF_uvy7:-KxIGBqVR9bNJTO4fFPe:b-qdtd05) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/create-live.component.scss#L33)):*
<details><summary><i>Previously, benstov wrote…</i></summary><blockquote>

We have discussed about it, and here it's ok to leave it as is (i understood the difference between that and leaving the /deep/ without parent)
</blockquote></details>

extract the nested forms styles to a separated file and include it as we are doing in theme.

---

*[src/applications/kmc-upload-app/prepare-entry/transcoding-profile-select/transcoding-profile-select.component.ts, line 89 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvvDjNlWfv-27sRkVfe:-KxIH7ko-RaEWR6aE0Nz:b32w0x) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/prepare-entry/transcoding-profile-select/transcoding-profile-select.component.ts#L89)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

you will fail right after the `.value` part....  
</blockquote></details>

OK

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/222)*
<!-- Sent from Reviewable.io -->

esakal commented:
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KxICCHI-Um5OUUPVBMK:-KxIe7A0wzkImD-3dxq1:b-cyrkft) (no related file):*
<details><summary><i>Previously, amirch1 (Amir Chervinsky) wrote…</i></summary><blockquote>

After creating a live entering, if the user chooses not to view the entry details, the screen is stuck on the floater and closing it results in "data lost" confirmation. Better to return to the entries list and refresh it so the user will be able to see the newly created entry
</blockquote></details>

+@benstov please check 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/222)*
<!-- Sent from Reviewable.io -->

benstov commented:
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KxICCHI-Um5OUUPVBMK:-KxIudG5_1kqBpT1MnkW:benwggn) (no related file):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

+@benstov please check 
</blockquote></details>

I couldn't reproduce this behavior

---

*[src/applications/kmc-upload-app/create-live/create-live.component.scss, line 33 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvuYPC-LQrBIwF_uvy7:-KxIuSKqQOQhgZWJalWT:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/create-live.component.scss#L33)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

extract the nested forms styles to a separated file and include it as we are doing in theme.
</blockquote></details>

Done.

---

*[src/applications/kmc-upload-app/create-live/create-live.component.ts, line 173 at r2](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KweGbQh6FFXYLbKQbKW:-KxIran24wY0rmmYv-Li:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/da4ac961a639b9a7ce1f20588573148bab9d5ba8/src/applications/kmc-upload-app/create-live/create-live.component.ts#L173)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

if you can remove the prefix '_' because this is considered as public api
</blockquote></details>

Done.

---

*[src/applications/kmc-upload-app/prepare-entry/prepare-entry.component.ts, line 31 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KxI7ubv0_6iAKvkFpcd:-KxIrgFkNaGPLEYMNyir:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/8e658724b3d8dce61fb2e99dd7d7cef996e521e2/src/applications/kmc-upload-app/prepare-entry/prepare-entry.component.ts#L31)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

isn't this a duplication with the line below?
</blockquote></details>

Done.

---

*[src/applications/kmc-upload-app/prepare-entry/prepare-entry.component.ts, line 48 at r3](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KxI8-edtEsroREE4MYC:-KxIrgz8gJcw63BHIyOE:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/8e658724b3d8dce61fb2e99dd7d7cef996e521e2/src/applications/kmc-upload-app/prepare-entry/prepare-entry.component.ts#L48)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

the same comment as above about the prefix of _
</blockquote></details>

Done.

---

*[src/applications/kmc-upload-app/prepare-entry/transcoding-profile-select/transcoding-profile-select.component.ts, line 89 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvvDjNlWfv-27sRkVfe:-KxIrmDuAO-U5J-PA9ar:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/prepare-entry/transcoding-profile-select/transcoding-profile-select.component.ts#L89)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

OK
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/222)*
<!-- Sent from Reviewable.io -->

benstov commented:
Review status: 30 of 37 files reviewed at latest revision, 5 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KxICCHI-Um5OUUPVBMK:-KxMSYgvNcNy-cdg1MgG:b-rl59hq) (no related file):*
<details><summary><i>Previously, benstov wrote…</i></summary><blockquote>

I couldn't reproduce this behavior
</blockquote></details>

actually i now understand what you meant, so do we wish to remove the cancel button inside the View entry details screen?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/222)*
<!-- Sent from Reviewable.io -->

benstov commented:
Review status: 30 of 37 files reviewed at latest revision, 5 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KxICCHI-Um5OUUPVBMK:-KxMXGz0oe7jJW2ozIfU:brnbs31) (no related file):*
<details><summary><i>Previously, benstov wrote…</i></summary><blockquote>

actually i now understand what you meant, so do we wish to remove the cancel button inside the View entry details screen?
</blockquote></details>

fixed. popup now closing upon rejecting entries list navigation

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/222)*
<!-- Sent from Reviewable.io -->

esakal commented:
Reviewed 7 of 7 files at r5.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[src/applications/kmc-upload-app/create-live/create-live.component.scss, line 33 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvuYPC-LQrBIwF_uvy7:-KyATsTJVOPzeZV7GvPj:bs646xz) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/create-live.component.scss#L33)):*
<details><summary><i>Previously, benstov wrote…</i></summary><blockquote>

Done.
</blockquote></details>

now that you have a dedicated file, shouldn't your remove this from here?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/222)*
<!-- Sent from Reviewable.io -->

benstov commented:
Review status: 26 of 37 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/applications/kmc-upload-app/create-live/create-live.component.scss, line 33 at r1](https://reviewable.io:443/reviews/kaltura/kmc-ng/222#-KvuYPC-LQrBIwF_uvy7:-KyBUBRS71RTUduUSOph:b-896fix) ([raw file](https://github.com/kaltura/kmc-ng/blob/88f48cdb19e807f1c5717d1335da15fe04267257/src/applications/kmc-upload-app/create-live/create-live.component.scss#L33)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

now that you have a dedicated file, shouldn't your remove this from here?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/222)*
<!-- Sent from Reviewable.io -->

esakal commented:
Reviewed 8 of 9 files at r6, 1 of 1 files at r7, 2 of 2 files at r8.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/kaltura/kmc-ng/222)*
<!-- Sent from Reviewable.io -->
